src/lib.rs

```diff
@@ -192,7 +65,7 @@ type Channel<T> = OneshotSender<Result<T, Error>>;
 /// Events used internally to communicate with the swarm, which is executed in the the background
 /// task.
 #[derive(Debug)]
-enum IpfsEvent {
+pub enum IpfsEvent {
```

Comment:
Why does this need to be public? Maybe `pub(crate)`?

Reply:
Doesn't compile otherwise, because the repo takes a `Sender<IpfsEvent>`, it's leaking private types.

Reply:
I'll fix this in a follow up when turning the repo in to a network behaviour. Then we should also be able to remove all the `Arc<Repo>`.

Reply:
Aah, so the real cause is Repo having `pub fn new(...) -> Self`? That'd would allow `pub(crate)` here as well.